### ISSUE TITLE
Critical failure on --#{$name}: var(--#{$target});

### ISSUE THREAD

**Issue by Xesenix** [sasstools/sass-lint#1187]
### What version of Sass Lint are you using?
v1.12.1

### Please include any relevant parts of your configuration
empty

### What did you do? Please include the actual source code causing the issue.
i have ran 
```bash
sass-lint
```
and after turning on verbose and removing surrounding code i identify this causes critical failure
```sass
@mixin define-css-color-variable($name, $color) {
	--#{$name}: var(--#{$target});
}
```

### What did you expect to happen?

Not throw critical error.

### What actually happened? Please include any error messages given to you by Sass Lint.

 I have got this error message:
```bash
    throw new exceptions.SassLintFailureError(errorCount.count + ' errors were detected in \n- ' + errorCount.files.join('\n- '));
    ^
SassLintFailureError: 1 errors were detected in
- src/styles/css-variables/mixins.scss
    at Function.sassLint.failOnError (#root_path#\node_modules\sass-lint\index.js:313:11)
    at detectPattern (#root_path#\node_modules\sass-lint\bin\sass-lint.js:31:10)
    at Object.<anonymous> (#root_path#\node_modules\sass-lint\bin\sass-lint.js:79:3)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
    at startup (bootstrap_node.js:190:16)
```

### If you're using a IDE plugin have you tried the CLI too?
Didn't use it as IDE plugin.
